google/resource_cloudfunctions_function.go

```diff
@@ -0,0 +1,462 @@
+		Delete: resourceCloudFunctionsDestroy,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("name", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"entry_point": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"memory": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Default:  FUNCTION_DEFAULT_MEMORY,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      DEFAULT_FUNCTION_TIMEOUT_IN_SEC,
+				ValidateFunc: validation.IntBetween(FUNCTION_TIMEOUT_MIN, FUNCTION_TIMEOUT_MAX),
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"storage_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"storage_object": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"trigger_bucket": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"trigger_http", "trigger_topic"},
+			},
+
+			"trigger_http": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeBool},
+				ConflictsWith: []string{"trigger_bucket", "trigger_topic"},
+			},
+
+			"trigger_topic": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"trigger_http", "trigger_bucket"},
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudFunctionsCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	service := config.clientCloudFunctions
```

Comment:
this is fine, but since you're only using the service variable once you may as well inline it into the request below (that'll match the rest of the resources in this repo as well)

Reply:
Fixed